docs/database-dynamodb-single-table.md

```diff
@@ -0,0 +1,92 @@
+
+## Quick start
+
+```yaml
+service: my-app
+provider:
+  name: aws
+
+constructs:
+    myTable:
+        type: database/dynamodb-single-table
+
+plugins:
+    - serverless-lift
+```
+
+On `serverless deploy`, a preconfigured DynamoDB table will be created.
+
+## How it works
+
+The `database/dynamodb-single-table` construct creates and configures the table for production:
+
+- a composite primary index with generic attributes names - `PK` for the partition key and `SK` for the sort key
+- 5 local secondary indexes, the maximum allowed, with generic names - `LSI-1` to `LSI-5` for the index names and `LSI-1-SK` to `LSI-5-SK` for the sort keys. Those indexes have no impact on pricing as long as their indexes are not populated with data. Modifying a table local secondary indexes configuration requires table re-creation, thus enabling all of them by default at table creation.
+- a configurable amount of up to 20 glocal secondary indexes with generic names - `GSI-1` to `GSI-20` for the index names, `GSI-1-PK` to `GSI-20-PK` for the partition keys and `GSI-1-SK` to `GSI-20-SK` for the sort keys
+- all indexes attributes have string data type, ideal for composite attribue - i.e. `value1#value2`
+- a DynamoDB stream publishing new and old values at each write operation on the table
+
+## Variables
+
+All database constructs expose the following variables:
+
+- `tableName`: the name of the deployed DynamoDB table
+- `tableStreamArn`: the ARN of the stream of the deployed DynamoDB table
+
+This can be used to inject the tableName to a Lambda functions using the SDK to read or write data from the table, for example:
+
+```yaml
+constructs:
+    myTable:
+        type: database/dynamodb-single-table
+
+functions:
+    myFunction:
+        handler: src/index.handler
+        environment:
+            TABLE_NAME: ${construct:myTable.tableName}
+```
+
+_How it works: the `${construct:myTable.tableName}` variable will automatically be replaced with a CloudFormation reference to the DynamoDB table._
+
+## Permissions
+
+By default, all the Lambda functions deployed in the same `serverless.yml` file **will be allowed to read/write into the table**.
+
+In the example below, there are no IAM permissions to set up: `myFunction` will be allowed to read and write into the `myTable` table.
+
+```yaml
+constructs:
+    myTable:
+        type: database/dynamodb-single-table
+
+functions:
+    myFunction:
+        handler: src/index.handler
+        environment:
+            TABLE_NAME: ${construct:myTable.tableName}
+```
+
+## Configuration reference
+
+### Global secondary indexes
+
+Global secondary indexes have a direct impact on the cost of a DynamoDB table. There is no GSI configured by default on the database construct.
+
+You can specify the amount of GSI you'd like to enable on a DynamoDB table using the `gsiQuantity` property.
+
+```yaml
+constructs:
+    myTable:
+        # ...
+        gsiQuantity: 3
```

Comment:
Nothing major but I'm wondering if `gsiCount` would work better? If any native English speaker can chime in here that would be great.

I'd say "count" is more widespread in developer language, but maybe "quantity" is used in DynamoDB docs/settings already?

Anyway, I'm fine either way, I'm just opening the topic just in case.

Reply:
I prefer `gsiCount` as well, updated it :)